akka-cluster/src/main/scala/akka/cluster/ClusterEvent.scala

```diff
@@ -278,6 +285,7 @@ object ClusterEvent {
         case (_, newMember :: oldMember :: Nil) if newMember.status != oldMember.status ⇒ newMember
       }
       val memberEvents = (newMembers ++ changedMembers) collect {
+        case m if m.status == Joining  ⇒ MemberJoined(m)
         case m if m.status == WeaklyUp ⇒ MemberWeaklyUp(m)
         case m if m.status == Up       ⇒ MemberUp(m)
         case m if m.status == Exiting  ⇒ MemberExited(m)
```

Comment:
We thought of joining as an internal detail also, I'd rather make it complete now and expose events for all state transitions. As a bonus we can make use of them in the ReadView so that JMX shows correct status.